middleware/kubernetes/k8sclient/k8sclient.go

```diff
@@ -0,0 +1,147 @@
+package k8sclient
+
+import (
+//    "fmt"
+    "net/url"
+)
+
+// API strings
+const (
+    api_base   = "/api/v1"
+    api_namespaces = "/namespaces"
+    api_services   = "/services"
+)
+
+// Defaults
+const (
+    default_baseurl = "http://localhost:8080"
+)
+
+
+type K8sConnector struct {
+    baseUrl string
+}
+
+func (self *K8sConnector) SetBaseUrl(u string) error {
+    valid_url, error := url.Parse(u)
+
+    if error != nil {
+        return error
+    }
+    self.baseUrl = valid_url.String()
+
+    return nil
+}
+
+func (self *K8sConnector) GetBaseUrl() string {
+    return self.baseUrl
+}
+
+
+func (self *K8sConnector) GetResourceList() *ResourceList {
+    resources := new(ResourceList)
+    
+    error := getJson((self.baseUrl + api_base), resources)
+    if error != nil {
+        return nil
+    }
+
+    return resources
+}
+
+
+func (self *K8sConnector) GetNamespaceList() *NamespaceList {
+    namespaces := new(NamespaceList)
+
+    error := getJson((self.baseUrl + api_base + api_namespaces), namespaces)
+    if error != nil {
+        return nil
+    }
+
+    return namespaces
+}
+
+
+func (self *K8sConnector) GetServiceList() *ServiceList {
+    services := new(ServiceList)
+
+    error := getJson((self.baseUrl + api_base + api_services), services)
+    if error != nil {
+        return nil
+    }
+
+    return services
+}
+
+
+func (self *K8sConnector) GetNamespaceNames() []string {
+    /*
+     * Return list of namespace names found in k8s.
+     */
+    var namespaces []string
+    return namespaces
+}
+
+
+func (self *K8sConnector) NamespaceExists(name string) bool {
+    /*
+     * Return true if namespace exists in k8s
+     */
+    var exists bool
```

Comment:
Just make these comments method doc comments, i.e.
// NameSpaceExists returns true if the namespace exists in k8s.